Gems/Atom/RPI/Code/Source/RPI.Public/Material/Material.cpp

```diff
@@ -99,24 +104,56 @@ namespace AZ
 
             MaterialPropertyFlags prevOverrideFlags = m_propertyOverrideFlags;
             AZStd::vector<MaterialPropertyValue> prevPropertyValues = m_propertyValues;
-
             // Initialize the shader runtime data like shader constant buffers and shader variants by applying the 
             // material's property values. This will feed through the normal runtime material value-change data flow, which may
             // include custom property change handlers provided by the material type.
             //
             // This baking process could be more efficient by doing it at build-time rather than run-time. However, the 
             // architectural complexity of supporting separate asset/runtime paths for assigning buffers/images is prohibitive.
             {
-                m_propertyValues.resize(materialAsset.GetPropertyValues().size());
+                m_propertyValues.resize(m_layout->GetPropertyCount());
                 AZ_Assert(m_propertyValues.size() == m_layout->GetPropertyCount(), "The number of properties in this material doesn't match the property layout");
 
-                for (size_t i = 0; i < materialAsset.GetPropertyValues().size(); ++i)
+                if (materialAsset.HasPropertyNames())
+                {
+                    const auto& propertyValues = materialAsset.GetPropertyValues();
+                    const auto& propertyNames = materialAsset.GetPropertyNames();
+                    const auto& propertyDescriptor = materialAsset.GetMaterialPropertiesLayout();
+                    const size_t latestPropertyValuesSize = propertyDescriptor->GetPropertyCount();
+                    AZStd::vector<bool> touchedIndices;
+                    touchedIndices.resize(latestPropertyValuesSize);
```

Comment:
You should construct this with the size and a default value of false.
Alternatively, this could be an unordered set of indexes.